chain/client/src/client_actor.rs

```diff
@@ -961,6 +961,7 @@ impl ClientActor {
             debug!(target: "client", "dropping block {} that is too far behind. Block height {} current tail height {}", block.hash(), block.header().height(), tail);
             return;
         }
+        self.client.chain.blocks_with_missing_chunks.prune_blocks_below_height(tail);
```

Comment:
this is not the best place to put it since invalid blocks can also trigger this. I suggest moving this to `on_block_accepted`

Reply:
Done in https://github.com/near/nearcore/pull/3710/commits/18842d1ee8ff975525c3ae883988d1417da5fde4